### PR TITLE
Support storyquestions atom

### PR DESCRIPTION
**Pull request by LATaylor-guardian**
## What does this change?
This change will render story question atoms when present on articles.

## What is the value of this and can you measure success?
This allows journalists to place questions they think may want answering by readers on articles and allows readers to upvote which of the questions they most want answering. This data will be fed into Ophan and allow journalists to make informed decisions about what content they commission off of the back of this information.

## Does this affect other platforms - Amp, Apps, etc?
Amp (I have excluded the atom from showing on AMP so that it doesn't fail the validation as it requires js. We will implement this on AMP at a later date).

## Screenshots
picture 42

Thanks @blongden73 for doing this!

## Tested in CODE?
No.

### CODE REVIEW

**Comment by LATaylor-guardian** [comment]
@NataliaLKB May you review please?

**Comment by NataliaLKB** [comment]
Does this conflict with the outbrain restrictions at all?

File: common/app/views/fragments/atoms/atom.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/atom.scala.html]: I think you want `amp = amp` here

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: How does this look with the submeta already in the page? Would you be able to add a screenshot?

**Comment by LATaylor-guardian** [comment]
@NataliaLKB What outbrain restrictions? 😆

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Could you use the `is-hidden` class instead https://github.com/guardian/frontend/blob/master/static/src/stylesheets/base/_helpers.scss#L63

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (LATaylor-guardian) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: picture 43

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Nit: for readability, could you have what is inside this link on a separate line. What about: 

```
<a href="#" id="@question.questionId" class="user__question-upvote js-ask-question-link">
    Ask this 
    @fragments.inlineSvg("tick", "icon")
</a>
```

Notice I removed the span as well because  the icon will already have a span with the class `tick` why next two of them?

File: static/src/stylesheets/module/atoms/_storyquestions.scss
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: A font mixin should be used here instead. Use one from here: https://github.com/guardian/frontend/blob/master/static/src/stylesheets/pasteup/typography/_src.scss

File: static/src/stylesheets/module/atoms/_storyquestions.scss
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: Does this span have a class? Best to use the class and not nest instead.

File: static/src/stylesheets/module/atoms/_storyquestions.scss
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: Best to add a more specific class to the svg. It will be added to the span and then the svg will be given the same class with `__svg` after it. For example, if the span has the class `user__tick-icon` or something similar, then the class on the svg would be `user__tick-icon__svg`. It is good to avoid nesting in css

File: static/src/stylesheets/module/atoms/_storyquestions.scss
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: Use the font mixin anytime you reference a font

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Thanks!

**Comment by NataliaLKB** [comment]
> What outbrain restrictions?

Good question! So we have a very strict contract with outbrain and if we don't follow the rules we don't get money from those page views. Putting containers above outbrain is an example of not following the rules. Best to talk to someone from @guardian/commercial-dev. Maybe @ScottPainterGNM ?

**Comment by LATaylor-guardian** [comment]
@NataliaLKB this is part of the content as it's an atom and is not a new container so I don't believe its a problem.

**Comment by NataliaLKB** [comment]
It is definitely worth double checking. Things like the email signup (even less container looking) is something that conflicted in the past. I just want to make sure that we won't have an unexpected loss of money first before we merge this.

**Comment by regiskuckaertz** [comment]
Hello - Natalia's right, _anything_ that pushes the Outbrain container further from the bottom of the article body is considered a breach of contract. In such cases, we have "non-compliant" Outbrain impressions, OB gives us less money for those, but we first need to estimate how many impressions we are talking about and the corresponding revenue loss. I would not merge without the approval of Catherine Murray in the AdOps team, which is ultimately the one deciding those matters.

**Comment by LATaylor-guardian** [comment]
@regiskuckaertz @NataliaLKB Thanks, we're speaking to Catherine Murray now about this.

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (gustavpursche) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: A link without an `href` should be a `<button>` :)

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (gustavpursche) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: > The P element represents a paragraph. It cannot contain block-level elements (including P itself). [source](https://www.w3.org/TR/html401/struct/text.html#h-9.3.1)

Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Should be a `div` in this case

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (gustavpursche) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: IMHO this should never be a `<h1>`, but only the article-headline, but I'm not sure how we deal with atoms and headline structure to be honest.

File: project/Dependencies.scala
Comment (NataliaLKB) [inline project/Dependencies.scala]: Have you checked a front loads fine running locally?

**Comment by gustavpursche** [comment]
@LATaylor-guardian Just a few thoughts on the layout:

- I wouldn't make the sub-headline so big. It's more of an explaining text, so maybe it could follow the paragraph styling from the article.
- Why is there a gray background behind the questions? It reduces the contrast and makes it harder to read.
- I'd make the question itself clickable. They are already blue and look somehow like a link to me. Alternative could be to not make them blue, but black-ish.
- I guess it would be easier, to move the `ask-this` button to the left, so it's easier to deal with the case of multiple lines? It would, maybe, make the borders obsolete, since thei are closer to the question.
- A nice effect for the "Voted state" could be: Only remove "Ask this" and make the "check" green.

Just wanted to get this out of my head :)

🌷

**Comment by LATaylor-guardian** [comment]
Thanks for your feedback @gustavpursche. I didn't do any of the design work on this so am simply happy for whomever is the design authority (and Its @blongden73 in this case) to make those decisions.

**Comment by blongden73** [comment]
@gustavpursche Thanks for your thoughts :-)

There are a couple of things with this. The headline style needs to feel like its not part of the article. I think keeping the size of the heading the same would make it feel like it was still within the article - esspecially as this is to sit above the social media buttons and tags etc. It need to be big enough especially at first to catch peoples eyes. 

If you are looking at just from the screenshots in the PR - it does make it harder to read but actually in the page it's fine. The thinking behind it is to highlight them to make it feel like these are questions that have been written - it adds a bit of visual interest it also makes them feel more important than if they were just given a treatment that blended into the style of the article.

If the ask this sat on the left you end up with either the ask this sitting in different places if it hangs of the end of the line. Or if it sits before the line starts its says ask this before you've read the question. 

I think you're right about the voted state having the tick remain would be better! 

:-)

**Comment by mchv** [comment]
Thanks @regiskuckaertz and @NataliaLKB! Following your useful comments we have asked nath. and chris w. to discuss this with the appropriate people and ensure to have the green light before it can get merged. We will keep you updated

File: static/src/stylesheets/module/atoms/_storyquestions.scss
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: This shouldn't pass the css linter. Mixins should go before declarations (at the top of a block)

Comment (blongden73) [inline static/src/stylesheets/module/atoms/_storyquestions.scss]: @NataliaLKB oooo my bad - yeah i got and error and fixed this maybe didnt do the second commit.

**Comment by LATaylor-guardian** [comment]
@NataliaLKB @regiskuckaertz The decision has been made to call non compliant on this. Moving forward are there any specific changes that need to be made to this PR in light of this? Many thanks.

**Comment by regiskuckaertz** [comment]
@LATaylor-guardian that seems like the best move going forward. You will have to flick the 'non-compliant' switch on, which @GHaberis knows all about as he's refactoring that right now in #16208

**Comment by GHaberis** [comment]
Hi @LATaylor-guardian, the checks which determine whether outbrain is compliant/non-compliant are defined in the check-mediator module and dispatched by the check-dispatcher module. A new check that returns whether this new component is present will need adding to these modules. As @regiskuckaertz mentioned I'm about to merge a PR which touches this logic so I'd hold until then.  I'll let you know when this is done, and am happy to assist.

**Comment by LATaylor-guardian** [comment]
Okay thanks @GHaberis, yes please let me know :)

**Comment by LATaylor-guardian** [comment]
@GHaberis Thank you very much for your help. Please see our changes for making outbrain non compliant. @NataliaLKB How is this looking now, are we good to go?

File: common/app/views/fragments/atoms/storyquestions.scala.html
Comment (NataliaLKB) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Nit: indentation is weird. Could we straighten this out?

Comment (LATaylor-guardian) [inline common/app/views/fragments/atoms/storyquestions.scala.html]: Done.

File: static/src/stylesheets/module/atoms/_storyquestions.scss~a50954f50b9fb4759c3c14b1246094d52299d760
Comment (NataliaLKB) [inline static/src/stylesheets/module/atoms/_storyquestions.scss~a50954f50b9fb4759c3c14b1246094d52299d760]: What is this file? I don't think this should be committed?

Comment (LATaylor-guardian) [inline static/src/stylesheets/module/atoms/_storyquestions.scss~a50954f50b9fb4759c3c14b1246094d52299d760]: Don't know how that got there - deleted.

**Review by NataliaLKB — APPROVED** [review]

File: common/app/model/Formats.scala
Comment (NataliaLKB) [inline common/app/model/Formats.scala]: cc @TBonnin to get your thoughts. This change would mean we don't have to repress fronts just to add a field for an atom.

I am happy with this change, but we have added it to our team's goals to understand and potentially come up with a better architecture for atoms in frontend, as adding it as part of facia seems a bit hacky.

File: common/app/model/Formats.scala
Comment (TBonnin) [inline common/app/model/Formats.scala]: Is that the correct parameter name? 😉

File: common/app/model/Formats.scala
Comment (TBonnin) [inline common/app/model/Formats.scala]: Would it make more sense to make the `Atoms` fields optional? 
That's the representation we are trying to model here?

File: common/app/model/Formats.scala
Comment (LATaylor-guardian) [inline common/app/model/Formats.scala]: Will fix now.

File: common/app/model/Formats.scala
Comment (LATaylor-guardian) [inline common/app/model/Formats.scala]: I don't mind - if you feel strongly about it then we can refactor in a later PR but I'd prefer to get this out as I need to move on.

Comment (TBonnin) [inline common/app/model/Formats.scala]: Yes. merge it. I don't feel strongly about it neither. I would like to explore making thos `Atoms` field optional in the future. I might miss the reason why it wouldn't make sense

Comment (LATaylor-guardian) [inline common/app/model/Formats.scala]: Great thanks! I think they are Seq's because an Option[Seq] feels odd when you can represent the None case as an empty Seq. It also makes working with them from there on in a little nicer.

Comment (TBonnin) [inline common/app/model/Formats.scala]: Agreed. In term of data, None would mean the field is not present, Seq.empty means field is present but doesn't contain any atom. (I give you that from the app perspective this is the same)
I don't think models should necessary match how they are represented in json. I am just throwing an idea (and its associated drawbacks) that you summarise well: Having to define and maintain our own `Reads` vs handling optional fields when working with `Atoms`.

Thanks for your work anyway. It looks like we can improve the way we support Atoms in general in the frontend codebase.

Comment (LATaylor-guardian) [inline common/app/model/Formats.scala]: Yeah sure, happy to help with this in any way I can.

**Pull request merged**
